modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -1357,23 +1387,30 @@ public ObjectEntry updateObjectEntry(
 			_objectDefinitionPersistence.findByPrimaryKey(
 				objectEntry.getObjectDefinitionId());
 
+		_validateServiceContextWorkflowAction(
+			objectDefinition.isEnableObjectEntryDraft(),
+			objectEntry.getStatus(), serviceContext.getWorkflowAction());
+
 		_validateValues(
 			user.isGuestUser(), objectEntry.getObjectDefinitionId(),
 			objectEntry, objectDefinition.getPortletId(), serviceContext,
 			userId, values);
 
 		Map<String, Serializable> transientValues = objectEntry.getValues();
 
+		int workflowAction = serviceContext.getWorkflowAction();
```

Comment:
Since you are using  `serviceContext.getWorkflowAction()` before, I think you should declare this variable before the first use (independently if the _validateServiceContextWorkflowAction() will throw an exception).